src/StoreKeeper/WooCommerce/B2C/Imports/AbstractProductImport.php

```diff
@@ -232,4 +236,42 @@ protected function cleanDotPath(string $shop_product_path): string
     {
         return trim($shop_product_path, '.');
     }
+
+    protected function processItem($dotObject, array $options = [])
+    {
+        $ShopModule = $this->storekeeper_api->getModule('ShopModule');
+
+        $shopProductId = $dotObject->get('id');
+        try {
+            $woocommerceProductId = $this->doProcessProductItem($dotObject, $options);
+
+            if (false !== $woocommerceProductId) {
+                $ShopModule->setShopProductObjectSyncStatusForHook([
+                    'status' => self::SYNC_STATUS_SUCCESS,
+                    'shop_product_id' => $shopProductId,
+                    'extra' => [
+                        'product_id' => $woocommerceProductId,
+                        'view_url' => get_permalink($woocommerceProductId),
+                        'edit_url' => admin_url('post.php?post='.$woocommerceProductId).'&action=edit',
+                        'date_synchronized' => (new \DateTime())->format('Y-m-d H:i:s'),
+                        'plugin_version' => implode(', ', [
+                            StoreKeeperOptions::PLATFORM_NAME.': '.get_bloginfo('version'),
+                            StoreKeeperOptions::VENDOR.' plugin: '.STOREKEEPER_WOOCOMMERCE_B2C_VERSION,
+                        ]),
+                    ],
+                ]);
+            }
+
+            return $woocommerceProductId;
+        } catch (\Throwable $throwable) {
+            $ShopModule->setShopProductObjectSyncStatusForHook([
+                'status' => self::SYNC_STATUS_FAILED,
+                'shop_product_id' => $shopProductId,
```

Comment:
Can you also check if you can still pass the data from extra? Are those always available?

If we can, please send it as well.
```

                        'product_id' => $woocommerceProductId,
                        'view_url' => get_permalink($woocommerceProductId),
                        'edit_url' => admin_url('post.php?post='.$woocommerceProductId).'&action=edit',
                        'date_synchronized' => (new \DateTime())->format('Y-m-d H:i:s'),
                        'plugin_version' => implode(', ', [
                            StoreKeeperOptions::PLATFORM_NAME.': '.get_bloginfo('version'),
                            StoreKeeperOptions::VENDOR.' plugin: '.STOREKEEPER_WOOCOMMERCE_B2C_VERSION,
                        ]),
```

Or atleast tell me what you can always send so I can make the others optional

Reply:
You may also need to adjust your unit test

Reply:
Added extra only if woocommerce product ID is available.
I have added a new method `setWooCommerceProductId` and `getWooCoomerceProductId` which will be called on catch clause and check if there is an available woocommerce product ID set.